### PR TITLE
[WIP] Feature/fix libmproxy 1269

### PR DESCRIPTION
**Pull request by m0sth8**
Issue #1269 
- [x] spiderman works with libmproxy
- [x] localproxy works with libmproxy
- [x] sqlmap works with libproxy
- [ ] Add more unittests for Proxy and w3afHandler
- [ ] Clean up.

### CODE REVIEW

**Comment by andresriancho** [comment]
@m0sth8 any plans to complete this PR? I would hate to see all this work wasted, and it would be the first step towards JS crawling support which is very much needed.

**Comment by m0sth8** [comment]
Hi, Andres!
I try to complete it in the next few days.



2015-01-01 23:22 GMT-03:00 Andres Riancho <notifications@github.com>:

**Comment by andresriancho** [comment]
Sounds great, thanks! Before you continue make sure you update your fork/branch with the latest code from master/develop.

Also remember that the pull request you send should be against the develop branch, and not master like this one is.

**Comment by andresriancho** [comment]
Been fighting against a couple of tests which were not passing, just to realize that it's because of https://github.com/andresriancho/w3af/issues/183 which would be fixed by this PR

File: w3af/core/controllers/ci/install_scripts/install_libmproxy_dependencies.sh [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash -x
+
+pip install pyOpenSSL==0.14
+pip install jinja2==2.7.3
+pip install -U --force-reinstall git+git://github.com/mitmproxy/netlib.git@55c2133b69bc39ad43c6ce1ab14b32019878e56a
+pip install  git+git://github.com/mitmproxy/mitmproxy.git@05a8c52f8f6c4fb5f1820475b9682da5a1eeadda
```
Comment (mhils) [inline w3af/core/controllers/ci/install_scripts/install_libmproxy_dependencies.sh]: `pip install mitmproxy` (with optional version specifier) is sufficient for installation.

**Comment by andresriancho** [comment]
@m0sth8 ping!

**Comment by andresriancho** [comment]
@m0sth8 time to be honest: Are you going to finish this PR? If not, I'll take over and finish it.

**Comment by andresriancho** [comment]
@m0sth8 sent me a private chat saying he would get this done

**Comment by andresriancho** [comment]
@m0sth8 please don't work on this PR, I'm taking over at #10429

**Pull request closed without merging**